docs/source/advanced/backends.rst

```diff
@@ -434,3 +528,131 @@ The following tables list the functions available in the `CUDA Runtime API <http
+
+  - Host/device signature must match in HIP(HCC)
+
+- A chain of functions must also provide correct host-device signatures
+
+  - E.g. a host function cannot be called from a host-device function
+
+- Recompile your target when HCC linker returned the error:
+  "File format not recognized
+  clang-7: error: linker command failed with exit code 1"
+- If compile-error occurred, the linker still may link, but without the device code
+- AMD device architecture currently hardcoded in ``alpakaConfig.cmake``
+
+Compiling HIP from Source
++++++++++++++++++++++++++
+
+Follow `HIP Installation`_ guide for installing HIP.
+HIP requires either *nvcc* or *hcc* to be installed on your system (see guide for further details).
+
+.. _HIP Installation: https://github.com/ROCm-Developer-Tools/HIP/blob/master/INSTALL.md
+
+- If you want the HIP binaries to be located in a directory that does not require superuser access, be sure to change the install directory of HIP by modifying the ``CMAKE_INSTALL_PREFIX`` cmake variable.
+- Also, after the installation is complete, add the following line to the ``.profile`` file in your home directory, in order to add the path to the HIP binaries to PATH: ``PATH=$PATH:<path_to_binaries>``
+
+.. code-block::
+
+   git clone --recursive https://github.com/ROCm-Developer-Tools/HIP.git
+   cd HIP
+   mkdir -p build
+   cd build
+   cmake -DCMAKE_BUILD_TYPE="${CMAKE_BUILD_TYPE}" -DCMAKE_INSTALL_PREFIX=${YOUR_HIP_INSTALL_DIR} -DBUILD_TESTING=OFF ..
+   make
+   make install
+
+- Set the appropriate paths (edit ``${YOUR_**}`` variables)
+
+.. code-block::
+
+  # HIP_PATH required by HIP tools
+  export HIP_PATH=${YOUR_HIP_INSTALL_DIR}
+  # Paths required by HIP tools
+  export CUDA_PATH=${YOUR_CUDA_ROOT}
+  # - if required, path to HCC compiler. Default /opt/rocm/hcc.
+  export HCC_HOME=${YOUR_HCC_ROOT}
+  # - if required, path to HSA include, lib. Default /opt/rocm/hsa.
+  export HSA_PATH=${YOUR_HSA_PATH}
+  # HIP binaries and libraries
+  export PATH=${HIP_PATH}/bin:$PATH
+  export LD_LIBRARY_PATH=${HIP_PATH}/lib64:${LD_LIBRARY_PATH}
+
+- Test the HIP binaries
+
+.. code-block::
+
+  # calls nvcc or hcc
+  which hipcc
+  hipcc -V
+  which hipconfig
+  hipconfig -v
+
+
+Verifying HIP Installation
+++++++++++++++++++++++++++
+
+- If PATH points to the location of the HIP binaries, the following command should list several relevant environment variables, and also the selected compiler on your ``system-\`hipconfig -f\```
+- Compile and run the `square sample`_, as pointed out in the original `HIP install guide`_.
+
+.. _square sample: https://github.com/ROCm-Developer-Tools/HIP/tree/master/samples/0_Intro/square
+.. _HIP install guide: https://github.com/ROCm-Developer-Tools/HIP/blob/master/INSTALL.md#user-content-verify-your-installation
+
+Compiling Examples with HIP Back End
+++++++++++++++++++++++++++++++++++++
+
+As of now, the back-end has only been tested on the NVIDIA platform.
```

Comment:
I don't think it's true, @psychocoderHPC ran it on AMD with HIP

Reply:
This is also a reason, why we have to overwork it. Since @tdd11235813 wrote the documentation there wars a least one release, which also solve some problems with 3D memory. I think, without CI, it is hard to keep the documentation up-to-date.